custom_components/localtuya/fan.py

```diff
@@ -58,12 +61,18 @@ def __init__(
         self._is_on = False
         self._speed = None
         self._oscillating = None
+        self._direction = None
 
     @property
     def oscillating(self):
         """Return current oscillating status."""
         return self._oscillating
 
+    @property
+    def current_direction(self) -> str:
```

Comment:
Typing is not correct as `self._direction` can be None (as initialized in the constructor). So it should be marked as `str | None` both here and in the constructor.

Reply:
I actually did this at first after reading through the FanEntity class in home-assistant core, but marking it like that currently causes ```TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'```. It seems that is resolved by adding ```from __future__ import annotations```, but that would be the first usage of ```__future__``` in this project and with Python not exactly being my strongest language I wasn't too sure about best-practice.

Typing is similarly implemented at the moment for ```self._speed``` and ```self._oscillating```, so I ultimately went with consistency there.

Happy to go either way.